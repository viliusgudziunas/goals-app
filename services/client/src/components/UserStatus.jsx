import React, { useEffect, useState } from 'react';
import axios from 'axios';

import './styles/UserStatus.css';

const UserStatus = () => {
  const [id, setId] = useState(null);
  const [email, setEmail] = useState('');

  useEffect(() => {
    const options = {
      url: `${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`,
      method: 'get',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${window.localStorage.authToken}`
      }
    };
    axios(options)
      .then(res => {
        setId(res.data.data.id);
        setEmail(res.data.data.email);
      })
      .catch(error => {
        console.log(error);
      });
  }, []);

  return (
    <>
      <div className='container'>
        <br />
        <h1>User Status</h1>
        <hr />
        <div className='row'>
          <div className='list-group'>
            <div className='list-group-item user-status-list-item'>
              <strong>{'User ID: '}</strong>
              {id}
            </div>
            <div className='list-group-item user-status-list-item'>
              <strong>{'Email: '}</strong>
              {email}
            </div>
          </div>
        </div>
      </div>
      <br />
      <br />
    </>
  );
};

export default UserStatus;
