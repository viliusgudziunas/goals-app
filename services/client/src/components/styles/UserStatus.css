.list-group-item {
  border: none;
  padding-top: 1px;
}
