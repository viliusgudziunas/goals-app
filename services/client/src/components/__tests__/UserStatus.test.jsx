import React from 'react';
import { shallow } from 'enzyme';

import UserStatus from '../UserStatus';

const listItems = [
  { index: 0, name: 'User ID' },
  { index: 1, name: 'Email' },
  { index: 2, name: 'Active' },
  { index: 3, name: 'Admin' }
];

describe('<UserStatus />', () => {
  const wrapper = shallow(<UserStatus />);
  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have a header', () => {
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper.find('h1').text()).toBe('User Status');
  });

  const list = wrapper.find('.list-group');
  it('should have a list', () => {
    expect(list.length).toBe(1);
  });

  listItems.forEach(listItem => {
    it(`should have a ${listItem.name} list item`, () => {
      const item = list.find('.list-group-item').get(listItem.index);
      expect(item.props.children[0].props.children).toContain(listItem.name);
    });
  });
});

// Should add another test to check that the fields get populated with values
// from state when the state becomes available
