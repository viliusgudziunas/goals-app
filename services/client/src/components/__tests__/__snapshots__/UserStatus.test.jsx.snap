// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStatus /> should match the snapshot 1`] = `"<div class=\\"container\\"><br/><h1>User Status</h1><hr/><div class=\\"row\\"><div class=\\"list-group\\"><div class=\\"list-group-item user-status-list-item\\"><strong>User ID: </strong></div><div class=\\"list-group-item user-status-list-item\\"><strong>Email: </strong></div></div></div></div>"`;
